platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/geometry/GeoJsonConversionTest.java

```diff
@@ -35,101 +38,94 @@
-        )
-      );
-      mapboxMap.getStyle().addLayer(new SymbolLayer("test-id", "test-id"));
-    }));
+    mapboxMap.getStyle().addSource(
+      new CustomGeometrySource("test-id",
+        new CustomProvider(fromFeatures(singletonList(fromGeometry(Polygon.fromLngLats(emptyList())))))
+      )
+    );
+    mapboxMap.getStyle().addLayer(new SymbolLayer("test-id", "test-id"));
   }
 
   @Test
+  @UiThreadTest
   public void testMultiPolygonFeatureCollection() {
     validateTestSetup();
-    onMapView().perform(getMapboxMapAction((uiController, mapboxMap) -> {
-      mapboxMap.getStyle().addSource(
-        new CustomGeometrySource("test-id",
-          new CustomProvider(fromFeatures(singletonList(fromGeometry(fromPolygon(Polygon.fromLngLats(emptyList()))))))
-        )
-      );
-      mapboxMap.getStyle().addLayer(new SymbolLayer("test-id", "test-id"));
-    }));
+    mapboxMap.getStyle().addSource(
+      new CustomGeometrySource("test-id",
+        new CustomProvider(fromFeatures(singletonList(fromGeometry(fromPolygon(Polygon.fromLngLats(emptyList()))))))
+      )
+    );
+    mapboxMap.getStyle().addLayer(new SymbolLayer("test-id", "test-id"));
   }
 
   @Test
+  @UiThreadTest
   public void testLineStringFeatureCollection() {
     validateTestSetup();
-    onMapView().perform(getMapboxMapAction((uiController, mapboxMap) -> {
-      mapboxMap.getStyle().addSource(
-        new CustomGeometrySource("test-id",
-          new CustomProvider(fromFeatures(singletonList(fromGeometry(fromLngLats(emptyList())))))
-        )
-      );
-      mapboxMap.getStyle().addLayer(new SymbolLayer("test-id", "test-id"));
-    }));
+    mapboxMap.getStyle().addSource(
+      new CustomGeometrySource("test-id",
+        new CustomProvider(fromFeatures(singletonList(fromGeometry(fromLngLats(emptyList())))))
+      )
+    );
+    mapboxMap.getStyle().addLayer(new SymbolLayer("test-id", "test-id"));
   }
 
   @Test
+  @UiThreadTest
   public void testMultiLineStringFeatureCollection() {
     validateTestSetup();
-    onMapView().perform(getMapboxMapAction((uiController, mapboxMap) -> {
-      mapboxMap.getStyle().addSource(
-        new CustomGeometrySource("test-id",
-          new CustomProvider(fromFeatures(singletonList(fromGeometry(fromLineString(fromLngLats(emptyList()))))))
-        )
-      );
-      mapboxMap.getStyle().addLayer(new SymbolLayer("test-id", "test-id"));
-    }));
+    mapboxMap.getStyle().addSource(
+      new CustomGeometrySource("test-id",
+        new CustomProvider(fromFeatures(singletonList(fromGeometry(fromLineString(fromLngLats(emptyList()))))))
+      )
+    );
+    mapboxMap.getStyle().addLayer(new SymbolLayer("test-id", "test-id"));
   }
 
+
   @Test
   public void testNegativeNumberPropertyConversion() {
     validateTestSetup();
-    onMapView().perform(getMapboxMapAction((uiController, mapboxMap) -> {
+    onView(isRoot()).perform(new MapboxMapAction((uiController, mapboxMap) -> {
```

Comment:
because it uses `uiController` needed [here](https://github.com/mapbox/mapbox-gl-native/pull/14051/files/2c45003f599b85279aad264bd7d0d5b9e2c3e334#diff-01090d147f18c53cf308d2797c54bef1R147), this PR doesn't fully exclude espresso from the codebase yet but that is next to tackle.